compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/diagnostics/LightTreePositioningStrategies.kt

```diff
@@ -170,6 +170,19 @@ object LightTreePositioningStrategies {
             return markElement(tree.operationReference(node) ?: node, tree)
         }
     }
+
+    val PARAMETER_DEFAULT_VALUE: LightTreePositioningStrategy = object : LightTreePositioningStrategy() {
+        override fun mark(node: LighterASTNode, tree: FlyweightCapableTreeStructure<LighterASTNode>): List<TextRange> {
+            return markElement(tree.findNextSiblingOfType(node, KtNodeTypes.TYPE_REFERENCE) ?: node, tree)
```

Comment:
Therefore, these two LT positioning strategies are my best guess. :\

At least, I can see what LT structure is used when the checker reports an error, and according to that, a parameter with a default value has two children: one is the type reference, and the other is the default value. So, the intention here is to find type reference for sure and then move to the next child.

I'm more confident on `vararg` modifier retrieval below, which seems more or less obvious. :)

Reply:
It's no longer a guess; I can step through the implementation and make sure the node of interest is chosen correctly.